book/src/storage/transfer-to-object.md

```diff
@@ -1,4 +1,132 @@
+public struct Receiving<phantom T: key> has drop {
+    id: ID,
+    version: u64,
+}
+
+/// Private receiving function for `key`-only objects.
+public fun receive<T: key>(parent: &mut UID, to_receive: Receiving<T>): T { /* ... */ }
+
+/// Public receiving function for `key + store` objects.
+public fun public_receive<T: key + store>(parent: &mut UID, to_receive: Receiving<T>): {
+  /* ... */
+}
+```
+
+The `Receiving` type is a special type that references an object-owned object. It cannot be
+constructed in Move, and can only be passed as a special input to a
+[Programmable Transaction Block (PTB)](../concepts/what-is-a-transaction.md). The `receive` (or
+`public_receive`) function must be called in order to exchange the `Receiving` instance for the
+actual object.
+
+While this may seem overwhelming, we will demonstrate the feature in a simple example.
+
+## Example
+
+To demonstrate the feature, let's consider a simple `PostOffice` object that can receive any objects
+transferred to it. Given that the transfer itself can be performed on the PTB level, or in a Move
+function, we will focus on the receiving part.
+
+```move
+module book::post_office;
+
+// While the `sui::transfer` module is implicitly imported, the `Receiving` type
+// is not. We need to import it explicitly.
+use sui::transfer::Receiving;
+
+/// The `PostOffice` object that can receive any objects transferred to it via
+/// a custom `receive` function.
+public struct PostOffice has key {
+    id: UID,
+}
+
+/// Voila! The `receive` function can now be called on the `PostOffice` object
+/// to receive any objects transferred to it.
+public fun public_receive<T: key + store>(
+    po: &mut PostOffice,
+    to_receive: Receiving<T>
+): T {
+    transfer::public_receive(&mut po.id, to_receive)
+}
+
+/// For objects without `store` (though, they would require special handling), we
+/// can implement the non-public version of the `receive` function.
+public fun receive<T: key>(po: &mut PostOffice, to_receive: Receiving<T>): T {
+    transfer::receive(&mut po.id, to_receive)
+}
+```
+
+In this example, we have defined a `PostOffice` object that can receive any objects transferred to
+it. We ommitted the creation of the `PostOffice` for brevity; additionally, an implementation like
+this would require some authorization to prevent unauthorized claims. We talk about authorization
+patterns in the [Advanced Programmability](./../programmability/) chapter.
+
+## Limitations
+
+Transfer to Object must be used with caution, as the recipient object must be implemented in a way
+that allows receiving. Worth noting, that objects owned by other objects must be taken by value, and
+cannot be borrowed as references, unlike objects owned by accounts.
+
+Transfer to Object is not the only way to create a relationship between objects. There are more
+flexible and feature-rich ways, such as the
+[Dynamic Fields](./../programmability/dynamic-fields.md), which we explore in the
```

Comment:
I do not know if this is the place for this comment, however as dynamic fields are referenced here, here it is:
Something that comes up a lot:
> Should I use dynamic fields or TTO?

One major difference is that one does not need to have ownership of the owner-object in order to transfer an object to it.
Dynamic fields on the other hand can be seen as having more "control" on what is added "below" the owner object.

Reply:
Been thinking about it a lot and still don't have an answer. Technically, in the book, it is too early to mention dynamic fields (they come mid next chapter), yet dynamic fields are too far from transfer to object. Very good point, and very much worth discussing!

Reply:
In my mind it's not really an either/or of TTO vs dynamic fields. TTO is about transfer, dynamic fields are about ownership -- i.e. it's perfectly valid to have a protocol where you use TTO to transfer something to an object, and then when it is received, it is turned into a dynamic field on the object it was received on.

Reply:
@amnn I agree that they're not replacing each other (though, in some situations they do), but more of a guide which one is suited for which purpose.